src/Raven.Server/Documents/ChangesClientConnection.cs

```diff
@@ -574,12 +583,83 @@ await using (var writer = new AsyncBlittableJsonTextWriter(context, ms))
                             break;
 
                         ms.TryGetBuffer(out ArraySegment<byte> bytes);
-                        await _webSocket.SendAsync(bytes, WebSocketMessageType.Text, true, _disposeToken);
+
+                        var sendTask = _webSocket.SendAsync(bytes, WebSocketMessageType.Text, true, _disposeToken);
+                        if (sendTask.IsCompleted)
+                        {
+                            await sendTask;
+                            continue;
+                        }
+
+                        if (await WaitForSendTaskAsync(sendTask, messagesCount, ms) == false)
+                            return;
                     }
                 }
             }
         }
 
+        private async Task<bool> WaitForSendTaskAsync(Task sendTask, int messagesCount, MemoryStream ms)
```

Comment:
This can _never_ return false, however?

Reply:
We can return false on dispose? Check 'ObjectDisposedException' handling inside.

Reply:
is there a reason not to throw there as well?

Reply:
changed